File: demo/src/main/java/org/example/demo/model/Message.java
package org.example.demo.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import java.sql.Timestamp;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "[message]")
public class Message {
    @Id
    @GeneratedValue(
            strategy= GenerationType.AUTO,
            generator="native"
    )
    @GenericGenerator(
            name = "native",
            strategy = "native"
    )
    @Column(name = "message_id")
    private Integer id;

    private String content;
    private Timestamp timestamp = new Timestamp(System.currentTimeMillis());

    @ManyToOne(cascade = {CascadeType.MERGE, CascadeType.REFRESH}, fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(cascade = {CascadeType.MERGE, CascadeType.REFRESH}, fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_id")
    private ChatSession chatSession;

    public Message(String messageContent, User user, ChatSession chatSession) {
        this.content = messageContent;
        this.user = user;
        this.chatSession = chatSession;
    }


    // Other message attributes

    // Constructors, getters, and setters
}

File: AnonyChat/Model/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AnonyChat.Model
{
    class Message
    {
        public int id { get; set; }
        public String content { get; set; }
        public DateTime timestamp = DateTime.Now ;
        public User user { get; set; }
        public ChatSession chatSession { get; set; }

        public Message(string content, User user, ChatSession chatSession)
        {
            this.content = content;
            this.user = user;
            this.chatSession = chatSession;
        }
        public Message()
        {

        }

    }
}
